src/mar/build/prepare.rs

```diff
@@ -0,0 +1,280 @@
+use aster::AstBuilder;
+
+use syntax::ast::{self, Expr, ExprKind, Pat,
+                  Block, SpannedIdent, Item};
+use syntax::fold::{self, Folder};
+use syntax::ptr::P;
+use syntax::codemap::{respan, Span};
+
+use mar::build::transition::ContainsTransition;
+
+pub fn prepare_item(item: P<Item>) -> P<Item> {
+    let mut expander = Expander::new();
+    let mut assigner = Assigner { next_node_id: 1 };
+
+    let expanded = fold::noop_fold_item_simple(item.unwrap(), &mut expander);
+    P(fold::noop_fold_item_simple(expanded, &mut assigner))
```

Comment:
These could be written a little more simply with: `expander.fold_item_simple(item.unwrap())`